sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs

```diff
@@ -479,6 +480,51 @@ public async Task RenameAsync()
             Response<PathProperties> response = await destDirectory.GetPropertiesAsync();
         }
 
+        [Test]
+        public async Task RenameAsync_FileSystemSAS()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem();
+
+            // Arrange
+            DataLakeDirectoryClient sourceDirectory = await test.FileSystem.CreateDirectoryAsync(GetNewDirectoryName());
+
+            DataLakeSasBuilder sasBuilder = new DataLakeSasBuilder
+            {
+                FileSystemName = test.FileSystem.Name,
+                Path = null,
+                Protocol = SasProtocol.None,
+                StartsOn = Recording.UtcNow.AddHours(-1),
+                ExpiresOn = Recording.UtcNow.AddHours(+1),
+                IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
+            };
+            sasBuilder.SetPermissions(DataLakeSasPermissions.All);
+            SasQueryParameters sourceSasQueryParameters = sasBuilder.ToSasQueryParameters(GetStorageSharedKeyCredentials());
+            string destDirectoryName = GetNewDirectoryName();
+            sourceDirectory = new DataLakeDirectoryClient(
+                new Uri(sourceDirectory.Uri.ToString() + "?" + sourceSasQueryParameters), GetOptions());
+
+            sasBuilder = new DataLakeSasBuilder
+            {
+                FileSystemName = test.FileSystem.Name,
+                Path = null,
+                Protocol = SasProtocol.None,
+                StartsOn = Recording.UtcNow.AddHours(-2),
+                ExpiresOn = Recording.UtcNow.AddHours(+2),
+                IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
+            };
+            sasBuilder.SetPermissions(DataLakeSasPermissions.All);
+            SasQueryParameters destSasQueryParameters = sasBuilder.ToSasQueryParameters(GetStorageSharedKeyCredentials());
```

Comment:
If we have created Dest Dir Client, can we user "destDataLakeDirectoryClient.Uri.PathAndQuery"?
Do you see any potential issue? 
If not, I will use this in PowerShell.

Reply:
Could you rephrase the question?

Reply:
I mean if I create a dest Directory client object, then get "destDataLakeDirectoryClient.Uri.PathAndQuery" and input to rename API as dest. 

In my testing, I see "PathAndQuery" will contain both path and the SAS, does it works? Not only for SAS, but also for other credential like sharedkey, Oauth.

Reply:
Currently, destination file system and destination path are 2 separate parameters.

A client created by Rename() using SAS can only be used with SAS going forward, since it was created with SAS, and clients are immutable.